ctest-next/src/tests.rs

```diff
@@ -28,64 +28,150 @@ extern "C" {
 }
 "#;
 
+macro_rules! collect_idents {
+    ($items:expr) => {
+        $items.iter().map(|a| a.ident()).collect::<Vec<_>>()
+    };
+}
+
 #[test]
 fn test_extraction_ffi_items() {
     let ast = syn::parse_file(ALL_ITEMS).unwrap();
 
     let mut ffi_items = FfiItems::new();
     ffi_items.visit_file(&ast);
 
+    assert_eq!(collect_idents!(ffi_items.aliases()), ["Foo"]);
+    assert_eq!(collect_idents!(ffi_items.constants()), ["bar"]);
+    assert_eq!(collect_idents!(ffi_items.foreign_functions()), ["malloc"]);
+    assert_eq!(collect_idents!(ffi_items.foreign_statics()), ["baz"]);
+    assert_eq!(collect_idents!(ffi_items.structs()), ["Array"]);
+    assert_eq!(collect_idents!(ffi_items.unions()), ["Word"]);
+}
+
+#[test]
+fn test_translation_type_ptr() {
+    let translator = Translator {};
+    let ty: syn::Type = syn::parse_str("*const *mut i32").unwrap();
```

Comment:
Suggestion: add something like this since it's repeated in every function

```rust
fn ty(s: &str) -> Result<String, ...> {
    let translator = Translator {};
    let ty: syn::Type = syn::parse_str("*const *mut i32").unwrap();
    translator.translate_type(&ty)
}
```

Also you probably only need one `array` function, one `pointer`, etc, and then those can have >1 assert. More compact and easier to add test cases that way.